app/main.py

```diff
@@ -1,23 +1,41 @@
 import init_django_orm  # noqa: F401
-
-from django.db.models import QuerySet
+from db.models import LoyaltyProgram, LoyaltyProgramParticipant
+from django.db.models import F, Q, QuerySet
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    queryset = LoyaltyProgram.objects.filter().values_list("name",
+                                                           "bonus_percentage")
+    return queryset
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    queryset = LoyaltyProgramParticipant.objects.filter(
+        last_activity__lt="2022-01-01",
```

Comment:
This filter may be simplified using `last_activity__year` parameter.

Reply:
I agree, simplify it